src/state_machine.cpp

```diff
@@ -104,24 +134,114 @@ void StateMachine::constructSofaMenu(State* from, State* next_obstacle)
       this->createGaitState(SOFA_SIT, SOFA_SIT_SELECTED, SOFA_SIT_ACTIVATED, "gait_sofa_sit", &sofa_standup);
 
   sofa_sit.backTo(from).withRight(&sofa_standup);
-  sofa_standup.backTo(&sofa_sit).withRight(&sofa_sit);
-
+  sofa_standup.backTo(from).withRight(&sofa_sit);
   from->withSelect(&sofa_sit);
 }
 
+void StateMachine::constructSlalomMenu(State* from, State* next_obstacle)
+{
+  State& slalom_walk = this->createGaitState(WALK_2O, WALK_2O_SELECTED, WALK_2O_ACTIVATED, "gait_walk", next_obstacle);
+
+  slalom_walk.backTo(from);
+
+  from->withSelect(&slalom_walk);
+}
+
+void StateMachine::constructRoughTerrainMenu(State* from, State* next_obstacle)
+{
+  State& small_step_rt_second = this->createGaitState(SMALLSTEP_RT_second, SMALLSTEP_RT_SELECTED,
+                                                      SMALLSTEP_RT_ACTIVATED, "gait_small_step", next_obstacle);
+  State& high_step_rt_second = this->createGaitState(HIGHSTEP_RT_second, HIGHSTEP_RT_SELECTED, HIGHSTEP_RT_ACTIVATED,
+                                                     "gait_high_step", &small_step_rt_second);
+  State& middle_step3_rt = this->createGaitState(MIDDLESTEP3_RT, MIDDLESTEP3_RT_SELECTED, MIDDLESTEP3_RT_ACTIVATED,
+                                                 "gait_third_middle_steps", &high_step_rt_second);
+  State& middle_step2_rt = this->createGaitState(MIDDLESTEP2_RT, MIDDLESTEP2_RT_SELECTED, MIDDLESTEP2_RT_ACTIVATED,
+                                                 "gait_second_middle_steps", &middle_step3_rt);
+  State& middle_step1_rt = this->createGaitState(MIDDLESTEP1_RT, MIDDLESTEP1_RT_SELECTED, MIDDLESTEP1_RT_ACTIVATED,
+                                                 "gait_first_middle_steps", &middle_step2_rt);
+  State& small_step_rt = this->createGaitState(SMALLSTEP_RT, SMALLSTEP_RT_SELECTED, SMALLSTEP_RT_ACTIVATED,
+                                               "gait_small_step", &middle_step1_rt);
+  State& high_step_rt = this->createGaitState(HIGHSTEP_RT, HIGHSTEP_RT_SELECTED, HIGHSTEP_RT_ACTIVATED,
+                                              "gait_high_step", &small_step_rt);  // de gait namen checken //
```

Comment:
Is this to do still relevant or is it already done?

Reply:
It is still relevant, but I will remove it here and make an issue in Jira